08_advanced/memory_snapshots/import_torch.py

```diff
@@ -0,0 +1,24 @@
+# # Memory snapshots for imports
+#
+# This program enables memory snapshotting for imports. We will create
+# a memory snapshot for this program after the import sequence is completed.
+# The program will start from the snapshot from there onwards.
+
+import modal
+
+image = modal.Image.debian_slim().pip_install("torch==2.2.1")
+stub = modal.Stub("import-torch", image=image)
+
+# All imports made up to this point will be included in the snapshot.
```

Comment:
This could be a bit misleading, because the import _after_ this point is also included in the snapshot. Could instead make an inline comment against each import

```python
import modal  # this import in global scope will be included in the snapshot

...
with image.imports():
    import torch  # this import inside the container image will also be included
...

```